client/oidc/src/main/java/io/quarkiverse/openapi/generator/oidc/providers/OAuth2AuthenticationProvider.java

```diff
@@ -31,12 +31,19 @@ public OAuth2AuthenticationProvider(String name,
 
     @Override
     public void filter(ClientRequestContext requestContext) throws IOException {
-        if (isTokenPropagation()) {
-            String bearerToken = getTokenForPropagation(requestContext.getHeaders());
-            bearerToken = sanitizeBearerToken(bearerToken);
-            requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, OidcConstants.BEARER_SCHEME + " " + bearerToken);
+        String bearerToken;
+
+        if (this.isTokenPropagation()) {
+            bearerToken = this.getTokenForPropagation(requestContext.getHeaders());
         } else {
             delegate.filter(requestContext);
```

Comment:
Why is filtering happening only on this branch? I'm concerned why our tests didn't catch this. :(

Reply:
what do you mean?

Reply:
Yes.

Reply:
Indeed, we are stopping the filtering chain there. Yes, you can leave it as it is now. I'll investigate it in the future.